pkg/kdump/kdump-view.jsx

```diff
@@ -20,166 +20,198 @@
 import '../lib/patternfly/patternfly-4-cockpit.scss';
 import cockpit from "cockpit";
 
-import React from "react";
+import React, { useState } from "react";
 import { Button } from "@patternfly/react-core/dist/esm/components/Button/index.js";
 import { Checkbox } from "@patternfly/react-core/dist/esm/components/Checkbox/index.js";
 import { Card, CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
-import { CodeBlockCode } from "@patternfly/react-core/dist/esm/components/CodeBlock/index.js";
 import { HelperText, HelperTextItem } from "@patternfly/react-core/dist/esm/components/HelperText/index.js";
 import { Flex } from "@patternfly/react-core/dist/esm/layouts/Flex/index.js";
 import { Form, FormGroup, FormSection } from "@patternfly/react-core/dist/esm/components/Form/index.js";
 import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect/index.js";
 import { Page, PageSection, PageSectionVariants } from "@patternfly/react-core/dist/esm/components/Page/index.js";
 import { DescriptionList, DescriptionListDescription, DescriptionListGroup, DescriptionListTerm } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal/index.js";
 import { Spinner } from "@patternfly/react-core/dist/esm/components/Spinner/index.js";
 import { Switch } from "@patternfly/react-core/dist/esm/components/Switch/index.js";
 import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput/index.js";
 import { Title } from "@patternfly/react-core/dist/esm/components/Title/index.js";
 import { Tooltip, TooltipPosition } from "@patternfly/react-core/dist/esm/components/Tooltip/index.js";
 import { OutlinedQuestionCircleIcon } from "@patternfly/react-icons";
 
+import { useDialogs, DialogsContext } from "dialogs.jsx";
 import { show_modal_dialog } from "cockpit-components-dialog.jsx";
```

Comment:
Is `show_modal_dialog` generally broken, or did the page just use it wrongly? We use it in over 20 other places, after all.

The crash test dialog still uses it, but that does not have any significant state (it's just a confirmation dialog).

Reply:
I find `show_modal_dialog` a bit too overengineered personally and harder to grasp.

As a follow up I plan to also rewrite the crash test dialog but this PR is already big enough.

Reply:
Ack. Thanks!